frontend/wallet/src/render/views/footer/SendButton.re

```diff
@@ -1,217 +1,54 @@
 open Tc;
 
-module Styles = {
-  open Css;
-
-  let contentContainer =
-    style([
-      display(`flex),
-      width(`percent(100.)),
-      flexDirection(`column),
-      alignItems(`center),
-      justifyContent(`center),
-    ]);
-};
-
-module ModalState = {
-  module Validated = {
-    type t = {
-      from: PublicKey.t,
-      to_: PublicKey.t,
-      amount: string,
-      fee: string,
-      memoOpt: option(string),
-    };
-  };
-  module Unvalidated = {
-    type t = {
-      fromStr: option(string),
-      toStr: string,
-      amountStr: string,
-      feeStr: string,
-      memoOpt: option(string),
-      errorOpt: option(string),
-    };
-  };
-};
-
-let emptyModal: option(PublicKey.t) => ModalState.Unvalidated.t =
-  activeWallet => {
-    fromStr: Option.map(~f=PublicKey.toString, activeWallet),
-    toStr: "",
-    amountStr: "",
-    feeStr: "",
-    memoOpt: None,
-    errorOpt: None,
-  };
-
-let validateInt64 = s =>
-  switch (Int64.of_string(s)) {
-  | i => i > Int64.zero
-  | exception (Failure(_)) => false
-  };
-
-let validatePubkey = s =>
-  switch (PublicKey.ofStringExn(s)) {
-  | k => Some(k)
-  | exception _ => None
-  };
-
-let validate:
-  ModalState.Unvalidated.t => Belt.Result.t(ModalState.Validated.t, string) =
-  state =>
-    switch (state, validatePubkey(state.toStr)) {
-    | ({fromStr: None}, _) => Error("Please specify a wallet to send from.")
-    | ({toStr: ""}, _) => Error("Please specify a destination address.")
-    | (_, None) => Error("Destination is invalid public key.")
-    | ({amountStr}, _) when !validateInt64(amountStr) =>
-      Error("Please specify a non-zero amount.")
-    | ({feeStr}, _) when !validateInt64(feeStr) =>
-      Error("Please specify a non-zero fee.")
-    | ({fromStr: Some(fromPk), amountStr, feeStr, memoOpt}, Some(toPk)) =>
-      Ok({
-        from: PublicKey.ofStringExn(fromPk),
-        to_: toPk,
-        amount: amountStr,
-        fee: feeStr,
-        memoOpt,
-      })
-    };
+module WalletLocked = [%graphql
+  {|
+    query walletLocked ($publicKey: PublicKey!) @bsRecord {
+      wallet(publicKey: $publicKey) {
+        locked @bsDecoder(fn: "Tc.Option.getExn")
```

Comment:
This should be fine as long as we only call this for pubkeys in ownedWallets, but I'm slightly worried someone will copy paste this and use it somewhere else at some point. Can we maybe add a comment or something above the module creation?

Reply:
Yeah I'll add a comment.